### PR TITLE
[Merged by Bors] - feat(Order/ConditionallyCompleteLattice): `lt_ciSup_iff` and variants

### PR DESCRIPTION
**Pull request by vihdzp**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean [hunk 1]
```diff
@@ -997,6 +997,24 @@ When `iInf f < a`, there is an element `i` such that `f i < a`.
 theorem exists_lt_of_ciInf_lt [Nonempty ι] {f : ι → α} (h : iInf f < a) : ∃ i, f i < a :=
   exists_lt_of_lt_ciSup (α := αᵒᵈ) h
 
+theorem lt_csSup_iff (hb : BddAbove s) (hs : s.Nonempty) : a < sSup s ↔ ∃ b ∈ s, a < b := by
+  rw [← not_iff_not]
+  simpa using csSup_le_iff hb hs
```
Comment (loefflerd) [inline Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:1002]:
```suggestion
  simpa only [not_le, not_forall₂, exists_prop] using (csSup_le_iff hb hs (a := a)).not
```

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean [hunk 2]
```diff
@@ -997,6 +997,24 @@ When `iInf f < a`, there is an element `i` such that `f i < a`.
...
+
+theorem csInf_lt_iff (hb : BddBelow s) (hs : s.Nonempty) : sInf s < a ↔ ∃ b ∈ s, b < a := by
+  rw [← not_iff_not]
+  simpa using le_csInf_iff hb hs
```
Comment (loefflerd) [inline Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:1006]:
```suggestion
  simpa only [not_le, not_forall₂, exists_prop] using (le_csInf_iff hb hs).not
```

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean [hunk 3]
```diff
@@ -997,6 +997,24 @@ When `iInf f < a`, there is an element `i` such that `f i < a`.
...
+
+theorem lt_ciSup_iff [Nonempty ι] {f : ι → α} (hb : BddAbove (range f)) :
+    a < iSup f ↔ ∃ i, a < f i := by
+  convert lt_csSup_iff hb (range_nonempty _)
+  simp_rw [mem_range, exists_exists_eq_and]
```
Comment (loefflerd) [inline Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:1011]:
```suggestion
  simpa only [mem_range, exists_exists_eq_and] using lt_csSup_iff hb (range_nonempty _)
```

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean [hunk 4]
```diff
@@ -997,6 +997,24 @@ When `iInf f < a`, there is an element `i` such that `f i < a`.
...
+
+theorem ciInf_lt_iff [Nonempty ι] {f : ι → α} (hb : BddBelow (range f)) :
+    iInf f < a ↔ ∃ i, f i < a := by
+  convert csInf_lt_iff hb (range_nonempty _)
+  simp_rw [mem_range, exists_exists_eq_and]
```
Comment (loefflerd) [inline Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:1016]:
```suggestion
  simpa only [mem_range, exists_exists_eq_and] using csInf_lt_iff hb (range_nonempty _)
```

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean [hunk 5]
```diff
@@ -1179,12 +1197,20 @@ theorem isLUB_csSup' {s : Set α} (hs : BddAbove s) : IsLUB s (sSup s) := by
...
   isLUB_le_iff (isLUB_csSup' hs)
 
 theorem csSup_le' {s : Set α} {a : α} (h : a ∈ upperBounds s) : sSup s ≤ a :=
   (csSup_le_iff' ⟨a, h⟩).2 h
 
+/-- In conditionally complete orders with a bottom element, the nonempty condition can be omitted
+from `lt_csSup_iff`. -/
+theorem lt_csSup_iff' (hb : BddAbove s) : a < sSup s ↔ ∃ b ∈ s, a < b := by
+  rw [← not_iff_not]
+  simpa using csSup_le_iff' hb
```
Comment (loefflerd) [inline Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:1212]:
```suggestion
  simpa only [not_le, not_forall₂, exists_prop] using (csSup_le_iff' hb).not
```

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean [hunk 6]
```diff
@@ -1209,13 +1235,21 @@ theorem exists_lt_of_lt_csSup' {s : Set α} {a : α} (h : a < sSup s) : ∃ b 
...
   (csSup_le_iff' h).trans forall_mem_range
 
 theorem ciSup_le' {f : ι → α} {a : α} (h : ∀ i, f i ≤ a) : ⨆ i, f i ≤ a :=
   csSup_le' <| forall_mem_range.2 h
 
+/-- In conditionally complete orders with a bottom element, the nonempty condition can be omitted
+from `lt_ciSup_iff`. -/
+theorem lt_ciSup_iff' {f : ι → α} (h : BddAbove (range f)) : a < iSup f ↔ ∃ i, a < f i := by
+  rw [← not_iff_not]
+  simpa using ciSup_le_iff' h
```
Comment (loefflerd) [inline Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:1251]:
```suggestion
    simpa only [not_le, not_forall] using (ciSup_le_iff' h).not
```

**Review by loefflerd — COMMENTED** [review]
Thanks for this nice contribution! I found a few small golfs using `Iff.not`.

File: Mathlib/Order/ConditionallyCompleteLattice/Basic.lean [hunk 7]
```diff
@@ -1663,4 +1697,4 @@ lemma iInf_coe_lt_top : ⨅ i, (f i : WithTop α) < ⊤ ↔ Nonempty ι := by
 end WithTop
 end WithTopBot
 
-set_option linter.style.longFile 1700
+set_option linter.style.longFile 1900
```
[the remark this answers is not in the document]
Comment (loefflerd) [reply Mathlib/Order/ConditionallyCompleteLattice/Basic.lean:1700]: I made a new PR #18029, based on top of this one, which splits the file into two smaller pieces by moving all the lemmas about indexed sup / inf to a separate file.

**Comment by loefflerd** [comment]
Looks like CI passes on this – are you happy to go ahead? If so, please remove the "awaiting author" tag and I'll flag it for maintainer-merging.

**Comment by loefflerd** [comment]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**